pkg/daemon/ceph/osd/volume.go

```diff
@@ -736,7 +736,9 @@ func callCephVolume(context *clusterd.Context, args ...string) (string, error) {
 	// failure log later without also printing out past failures
 	// TODO: does this mess up expectations from the ceph log collector daemon?
 	logPath := "/tmp/ceph-log"
-	os.MkdirAll(logPath, 0644)
+	if err := os.MkdirAll(logPath, 0644); err != nil {
+		logger.Errorf("failed to create dir %q. %v", logPath, err)
```

Comment:
In most of these occurrences, we will want to return the error instead of just log it, for example:
```suggestion
		return "", errors.Wrapf(err, "failed to create dir %q", logPath)
```